controllers/managedmcg_controller.go

```diff
@@ -999,3 +1005,15 @@ func (r *ManagedMCGReconciler) ensureConsolePlugin(clusterVersion string) error
 
 	return nil
 }
+
+func determineOpenShiftVersion(client client.Client) (string, error) {
+	clusterVersionList := configv1.ClusterVersionList{}
+	if err := client.List(context.TODO(), &clusterVersionList); err != nil {
```

Comment:
Reason to just have this function ot attached to any struct is that this can be generally called to fetch cluster version.

> can we get context
we dont actually do anything with context, So passing empty context that makes it readable as well. Can also pass `ctx` but becomes ambiguos. Please suggest

>  clinet from ManagedMCGReconciler
passing the same

Reply:
I think if the context is available we should use this, if you are sure we are not using the context inside the method List() we can pass empty context

Reply:
done